stages/00_download.py

```diff
@@ -0,0 +1,68 @@
+from bs4 import BeautifulSoup
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import logging
+from pathlib import Path
+import re
+import requests
+import shutil
+
+logging.basicConfig(format="%(asctime)s %(message)s")
+logger = logging.getLogger(__name__)
+logger.setLevel("INFO")
+
+
+def create_download_dir():
+    logger.info("Creating download/ directory")
+    downloads = Path("download")
+    downloads.mkdir(exist_ok=True)
+
+
+URL = "https://downloads.thebiogrid.org/BioGRID/Release-Archive/BIOGRID-4.4.232/"
+
+
+def get_biogrid_html():
+    logger.info("Fetching BioGRID HTML page")
+    biogrid_downloads = requests.get(URL, headers={"accept": "text/html"})
+    html = biogrid_downloads.text
+    soup = BeautifulSoup(html, features="html.parser")
+    return soup
+
+
+tab3_files = re.compile("https://.*/Download/(?!.*OSPREY).*\.tab3.zip$")
```

Comment:
Looks good so far, but we need more than the `tab3` format. These are the files that we should get:

```
BIOGRID-ALL-4.4.232.tab3.zip
BIOGRID-CHEMICALS-4.4.232.chemtab.zip
BIOGRID-CORONAVIRUS-4.4.232.tab3.zip
BIOGRID-IDENTIFIERS-4.4.232.tab.zip
BIOGRID-MV-Physical-4.4.232.tab3.zip
BIOGRID-ORGANISM-4.4.232.tab3.zip
BIOGRID-PROJECT-autophagy_project-4.4.232.zip
BIOGRID-PROJECT-covid19_coronavirus_project-4.4.232.zip
BIOGRID-PROJECT-fanconi_anemia_project-4.4.232.zip
BIOGRID-PROJECT-glioblastoma_project-4.4.232.zip
BIOGRID-PROJECT-kinome_project_sc-4.4.232.zip
BIOGRID-PROJECT-ups_project-4.4.232.zip
BIOGRID-PTMS-4.4.232.ptm.zip
BIOGRID-SYSTEM-4.4.232.tab3.zip
```

diff with the current download:
```diff
--- b	2024-04-18 13:57:54.112292341 -0400
+++ a	2024-04-18 13:55:49.854351188 -0400
@@ -1,5 +1,14 @@
 BIOGRID-ALL-4.4.232.tab3.zip
+BIOGRID-CHEMICALS-4.4.232.chemtab.zip
 BIOGRID-CORONAVIRUS-4.4.232.tab3.zip
+BIOGRID-IDENTIFIERS-4.4.232.tab.zip
 BIOGRID-MV-Physical-4.4.232.tab3.zip
 BIOGRID-ORGANISM-4.4.232.tab3.zip
+BIOGRID-PROJECT-autophagy_project-4.4.232.zip
+BIOGRID-PROJECT-covid19_coronavirus_project-4.4.232.zip
+BIOGRID-PROJECT-fanconi_anemia_project-4.4.232.zip
+BIOGRID-PROJECT-glioblastoma_project-4.4.232.zip
+BIOGRID-PROJECT-kinome_project_sc-4.4.232.zip
+BIOGRID-PROJECT-ups_project-4.4.232.zip
+BIOGRID-PTMS-4.4.232.ptm.zip
 BIOGRID-SYSTEM-4.4.232.tab3.zip
```